tests/src/KeyManagerTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Provides \Drupal\Tests\key\KeyManagerTest.php
+ */
+
+namespace Drupal\Tests\key;
+
+use Drupal\key\Plugin\KeyType\SimpleKey;
+use Drupal\key\Entity\Key;
+use Drupal\key\KeyManager;
+
+class KeyManagerTest extends KeyTestBase {
```

Comment:
Yes, there is some overlap between functional and unit tests, but there are some differences.

1. Unit tests test the individual class in isolation without having to worry about state. A unit test should assert valid output for a method for each code path.
   1. This also means that unit tests are faster because it does not depend on system state. This is hard to do in practice in Drupal 8 because of the amount of mocking that has to be done, but those tests ran in less than in a second. :+1: 
2. Functional tests are good for testing the output from an actual browser, but the downside is that it takes longer to run because we need to spin up a database test.

I think as Drupal 8.1.x evolves there will be less functional tests when unit tests will suffice. This would reduce the testing infrastructure burden and speed up feedback on patches.

The phpunit test directory standard is to be in `tests` and the SimpleTest directory in `src/tests`. This is seen in Core as well.